frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx

```diff
@@ -49,9 +69,33 @@ const NamespaceStoreForm: React.FC<NamespaceStoreFormProps> = withHandlePromise<
     errorMessage,
     handlePromise,
     redirectHandler,
+    isPage,
     namespace,
   } = props;
 
+  const [isSubmit, setIsSubmit] = React.useState(false);
+  let isPhaseDefined = false;
+  const nooBaaNsResource: WatchK8sResource = React.useMemo(() => {
+    const storeName = isSubmit ? nsName : null;
+    return {
+      kind: referenceForModel(NooBaaNamespaceStoreModel),
+      namespaced: true,
+      isList: false,
+      name: storeName,
+      namespace,
+    };
+  }, [namespace, nsName, isSubmit]);
+  const [newNnsObj, newNnsLoaded, newNnsErr] = useK8sWatchResource<NamespaceStoreKind>(
+    nooBaaNsResource,
+  );
+
+  if (newNnsLoaded && !newNnsErr) {
+    isPhaseDefined = newNnsObj?.status?.phase !== undefined;
+  }
+  if (isSubmit && isPhaseDefined) {
+    redirectHandler();
+  }
+
```

Comment:
Sam issue what i saw in create-bs.tsx

1. When I use the already existing backing store name, the modal won't throw any error. Line 98 will assume a new pool is created and close the modal. User may assume this backing store is created with new inputs.

2. Any failure that happened while creating a backing store and backing store is not created, this page will keep Invoke LoadingComponent. 
    * Resolve this by onError parameter of handlePromise or errorMessage  (Either you need to display error message or invoke close :) )

3. If the backing store is created but if it is not ready (e.g ReconcileFailed), the module will close without any intimation to the user.

Reply:
for more info please check 
https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/create-block-pool-modal.tsx#L90

https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/create-block-pool-modal.tsx#L66